### PR TITLE
Feature/shareable md link

### PR DESCRIPTION
**Pull request by twalton83**
- added share feature for the lesson previewer
- converts Lesson Previewer component content to base64 and appends it to a string comprised of the `window` host and pathname
- tested for accuracy of conversion and link 

**Usage**
New share button
Screen Shot 2021-01-10 at 9 30 26 AM
Share button shows "Copied!" for 3 seconds 
Screen Shot 2021-01-10 at 9 30 30 AM

ezgif com-gif-maker (2)

Related issue: #1576

### CODE REVIEW

**Comment by I3uckwheat** [comment]
@twalton83 Can you add some screenshots and gifs to show what these changes do?

**Comment by twalton83** [comment]
@I3uckwheat Updated the OP!

File: app/javascript/src/js/mdPreviewShare.test.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { generateLink, encodeContent, decodeContent } from './mdPreviewShare';
+
+describe('encodeContent', () => {
+  it('content is correctly encoded to base64', () => {
+    const buffer = Buffer.from('This is a test string!', 'utf-8');
+    expect(encodeContent('This is a test string!')).toBe(buffer.toString('base64'));
+  });
+  it('content including numbers is correctly encoded to base64', () => {
```
Comment (KevinMulhern) [inline app/javascript/src/js/mdPreviewShare.test.js:8]: Nitpick, can we separate each test with a new line please? just to make it easier a little easier to read.

File: app/javascript/src/js/mdPreviewShare.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+function generateLink(content) {
```
Comment (KevinMulhern) [inline app/javascript/src/js/mdPreviewShare.js:1]: Another minor nitpick 😄  can we rename this file from from `mdPreviewShare.js` to `previewShare.js` please? we could potentially reuse these methods for any sort of content.

**Review by KevinMulhern — CHANGES_REQUESTED** [review]
I've left a couple of small comments @twalton83  but overall this looks brilliant. Thanks for all your work on it.

File: app/javascript/src/js/mdPreviewShare.test.js (hunk 1 above)
Comment (twalton83) [reply app/javascript/src/js/mdPreviewShare.test.js:8]: Of course! Still kind of getting used to spacing conventions :)

File: app/javascript/src/js/mdPreviewShare.js (hunk 2 above)
Comment (twalton83) [reply app/javascript/src/js/mdPreviewShare.js:1]: You got it!

**Comment by twalton83** [comment]
@KevinMulhern updated things :)

File: app/javascript/components/lesson-preview/index.jsx [hunk 3]
```diff
@@ -22,10 +25,30 @@ const LessonPreview = () => {
...
+    const encodedContent = encodeContent(content);
+    const link = generateLink(encodedContent);
+    navigator.clipboard.writeText(link).then(() => setCopied(true));
+  };
+
   useEffect(() => {
     Prism.highlightAll();
   }, [convertedContent]);
 
+  useEffect(() => {
```
Comment (I3uckwheat) [inline app/javascript/components/lesson-preview/index.jsx:38]: @KevinMulhern Do we need to also test the component itself?

Comment (KevinMulhern) [reply app/javascript/components/lesson-preview/index.jsx:38]: Yeah good shout, we can add unit tests around the button text and copying to clip board behaviour.

Comment (twalton83) [reply app/javascript/components/lesson-preview/index.jsx:38]: I'll go ahead and figure that out then :)

**Review by KevinMulhern — APPROVED** [review]
Great work @twalton83, this makes the preview feature so much more usable 🚀

**Pull request merged**